### DISCUSSION TITLE
import_map root-relative path convention

### DISCUSSION THREAD

**Discussion by TreetopFlyer** [alephjs/aleph.js#87] [Q&A] [answered]
I'm probably making a silly mistake but figured I'd ask. To get around long-winded "../" in import statements in deeply nested pages e.g. `import Logo from '../../../components/logo.tsx'`
I tried modifying import_map.json to be
```
{
    "imports":
    {
        "~/":"./components/"
    }
}
```
(as is similarly indicated at the bottom of [https://deno.land/manual/linking_to_external_code/import_maps](https://deno.land/manual/linking_to_external_code/import_maps))
so that I could just `import Logo from '~/logo.tsx'` wherever I want. However this produces `error: Uncaught (in promise) NotFound: The system cannot find the path specified. (os error 3)` when running the Aleph project.

If instead I were to use the above import_map.json on a vanilla deno project (not an Aleph project, but using the same folder structure for the sake of example) with `deno run --import-map=import_map.json...` any imports prefixed with "~/" work correctly.

**Comment by ije** [comment, marked as answer]
hi @TreetopFlyer, **v0.3** will support this!

**Reply by TreetopFlyer** [reply]
@ije very cool! Aleph is such an impressive project can't wait to see where it goes.

**Reply by ije** [reply]
i was blocked for a bit time, since deno bundler(swc) can not work fine currently (including 1.6.3).
